"restricted type `DYNAMIC_TIME_ZONE_INFORMATION` in public interface" with `v0.2.26`

Hi,

I noticed in our [CI](https://github.com/uutils/parse_datetime/pull/297) that the new release (`v0.2.26`) fails on `windows-latest` with Rust `1.71.1` with the following error:
```
error[E0446]: restricted type `DYNAMIC_TIME_ZONE_INFORMATION` in public interface
  --> C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\jiff-0.2.26\src\tz\system\windows\ffi.rs:35:1
   |
4  | pub(super) struct DYNAMIC_TIME_ZONE_INFORMATION {
   | ----------------------------------------------- `DYNAMIC_TIME_ZONE_INFORMATION` declared as restricted
...
35 | windows_link::link!("kernel32.dll" "system" fn GetDynamicTimeZoneInformation(info: *mut DYNAMIC_TIME_ZONE_INFORMATION) -> u32);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak restricted type
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -355,6 +355,19 @@ jobs:
     # - name: Run integration tests
       # run: cargo test --test integration
 
+  # Same as above, but for Windows. The MSRV is technically higher because Jiff
+  # inherits the MSRV of system-level dependencies (like `windows-link`).
+  msrv-win:
+    runs-on: windows-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v5
+    - name: Install Rust
+      uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: 1.71.1
+    - run: cargo build --verbose
+
   # This job tests that examples outside the workspace build.
   #
   # These are outside the workspace because their dependency graphs are

--- a/src/tz/system/windows/ffi.rs
+++ b/src/tz/system/windows/ffi.rs
@@ -1,7 +1,12 @@
-pub(super) const TIME_ZONE_ID_INVALID: u32 = 0xffffffff;
+pub const TIME_ZONE_ID_INVALID: u32 = 0xffffffff;
 
+// MSRV(??): Must be marked as public to avoid errors in older Rust compilers.
+// Rust 1.71 needs this. But Rust 1.95 does not. I'm not sure where the cross
+// over point is, but it'd be nice to tighten this up a bit. In any case, this
+// isn't actually re-exported anywhere in Jiff's public API, so it's fine.
+// ---AG
 #[repr(C)]
-pub(super) struct DYNAMIC_TIME_ZONE_INFORMATION {
+pub struct DYNAMIC_TIME_ZONE_INFORMATION {
     bias: i32,
     standard_name: [u16; 32],
     standard_date: SYSTEMTIME,
@@ -9,8 +14,8 @@ pub(super) struct DYNAMIC_TIME_ZONE_INFORMATION {
     daylight_name: [u16; 32],
     daylight_date: SYSTEMTIME,
     daylight_bias: i32,
-    /// This is the only field we actually need.
-    pub(super) timezone_key_name: [u16; 128],
+    // This is the only field we actually need.
+    pub timezone_key_name: [u16; 128],
     dynamic_daylight_time_disabled: u8,
 }
 

